src/java.base/unix/native/libjli/java_md.c

```diff
@@ -296,6 +296,16 @@ CreateExecutionEnvironment(int *pargc, char ***pargv,
                            char jrepath[], jint so_jrepath,
                            char jvmpath[], jint so_jvmpath,
                            char jvmcfg[],  jint so_jvmcfg) {
+    /* Compute/set the name of the executable. This is needed for macOS. */
+    SetExecname(*pargv);
```

Comment:
Why is `SetExecname()` needed for the static case?

Reply:
Because of how macOS re-exec the launcher to reserve the main thread for GUI operations. Please check the rather extensive documentation elsewhere in this file.